From d454ec3cd0bcc3f5dcff73023061b4cf0c8a241b Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Thu, 26 Sep 2019 13:57:08 -0400
Subject: [PATCH 25/26] oxt: flask: Trim create_domain_common domain perms

Remove some permissions from create_domain_common to bring it into
alignment with OpenXT's policy.
---
 tools/flask/policy/modules/xen.if | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index 11395c61f6..8d78e0cf73 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -51,15 +51,13 @@ define(`create_domain_common', `
 			getvcpuinfo getaddrsize getaffinity setaffinity
 			settime setdomainhandle getvcpucontext set_misc_info };
 	allow $1 $2:domain2 { set_cpuid settsc setscheduler setclaim
-			set_vnumainfo get_vnumainfo cacheflush
-			psr_cmt_op psr_alloc soft_reset
-			resource_map get_cpu_policy };
+			resource_map };
 	allow $1 $2:security check_context;
 	allow $1 $2:shadow enable;
 	allow $1 $2:mmu { map_read map_write adjust memorymap physmap pinpage mmuext_op updatemp stat };
 	allow $1 $2:grant setup;
 	allow $1 $2:hvm { getparam hvmctl gethvmc sethvmc
-			setparam nested altp2mhvm altp2mhvm_op dm };
+			setparam nested altp2mhvm dm };
 ')
 
 # create_domain(priv, target)
-- 
2.21.0

